src/services/mongoose/wallet-invoices.ts

```diff
@@ -31,15 +31,10 @@ export const WalletInvoicesRepository = (): IWalletInvoicesRepository => {
   const update = async ({
     paymentHash,
     paid,
-  }: {
-    paymentHash: PaymentHash
-    paid: boolean
-  }): Promise<WalletInvoice | RepositoryError> => {
+  }: WalletInvoice): Promise<WalletInvoice | RepositoryError> => {
     try {
       const data = { paid }
-      const invoiceUser = await InvoiceUser.findOneAndUpdate({ _id: paymentHash }, data, {
-        new: true,
-      })
+      const invoiceUser = await InvoiceUser.findOneAndUpdate({ _id: paymentHash }, data)
       if (!invoiceUser) {
         return new RepositoryError("Couldn't update invoice for payment hash")
       }
```

Comment:
we need the new here. otherwise the data returned back the line below will be the old record

Reply:
https://mongoosejs.com/docs/tutorials/findoneandupdate.html#getting-started 
```
You should set the new option to true to return the document after update was applied.
```